catalog/spatial/csw/spatial-csw-common/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/common/GetRecordsRequest.java

```diff
@@ -379,11 +379,17 @@ public GetRecordsType get202RecordsType() throws CswException {
                 queryConstraint.setCqlText(getConstraint());
             } else if (getConstraintLanguage().equalsIgnoreCase(CswConstants.CONSTRAINT_LANGUAGE_FILTER)) {
                 try {
+                    XMLInputFactory xmlInputFactory = XMLInputFactory.newFactory();
+                    xmlInputFactory.setProperty(XMLInputFactory.IS_SUPPORTING_EXTERNAL_ENTITIES,
+                            false);
+                    xmlInputFactory.setProperty(XMLInputFactory.SUPPORT_DTD, false);
+                    XMLStreamReader xmlStreamReader =
+                            xmlInputFactory.createXMLStreamReader(new StringReader(constraint));
+
                     Unmarshaller unmarshaller = JAX_BCONTEXT.createUnmarshaller();
```

Comment:
@stustison: While I fully agree with your statement, I would argue that changing the unmarshaller over to the platform-parser-api would be out of scope for this ticket and could potentially make it harder to backport to 4.2.1 as needed. Maybe we should just create a ticket to address this later?